src/main/java/Rica.java

```diff
@@ -0,0 +1,154 @@
+            return null;
+        }
+        Task selectedTask = Rica.getPastTexts().get(indexOfTask);
+        if (selectedTask.getIsDone()) {
+            printlnWithIndent(" Take a break maybe? Alright marked as done my friend:");
+            printlnWithIndent("    " + selectedTask);
+            return selectedTask;
+        }
+        Rica.getPastTexts().remove(indexOfTask);
+        selectedTask = selectedTask.setDone();
+        Rica.getPastTexts().add(indexOfTask, selectedTask);
+        printlnWithIndent(" Shall remember that this task as done:");
+        printlnWithIndent("    " + selectedTask);
+        return selectedTask;
+    }
+
+    /**
+     * Mark a given task in the task list as not done
+     *
+     * @param indexOfTask Index of desired task in the task list
+     * @return Task object representing the desired task being marked as not done
+     */
+    private static Task unmarkDone(int indexOfTask) {
+        if (indexOfTask < 0 || indexOfTask >= Rica.getPastTexts().size()) {
+            printlnWithIndent(" You alright? I don't think that task exists xD");
+            return null;
+        }
+        Task selectedTask = Rica.getPastTexts().get(indexOfTask);
+        if (!selectedTask.getIsDone()) {
+            printlnWithIndent(" Getting a little ahead of yourself are you xD It's not even done:");
+            printlnWithIndent("    " + selectedTask);
+            return selectedTask;
+        }
+        Rica.getPastTexts().remove(indexOfTask);
+        selectedTask = selectedTask.setNotDone();
+        Rica.getPastTexts().add(indexOfTask, selectedTask);
+        printlnWithIndent(" (Why??) Anyway, I've marked this task as not done yet:");
+        printlnWithIndent("    " + selectedTask);
+        return selectedTask;
+    }
+
+    private static boolean hasAnyTasks() {
+        return !Rica.getPastTexts().isEmpty();
+    }
+
+    public static void main(String[] args) {
+        // Define horizontal line in a variable since it's printed multiple times
+
+        printlnWithIndent(Rica.LINE);
+        printlnWithIndent(" Hello! I'm R.I.C.A.");
+        printlnWithIndent((" That's Really-Intelligent-Chat-Assistant for you!"));
+        printlnWithIndent(" How may I be of assistance?");
+        printlnWithIndent(Rica.LINE);
+        Rica.runCommands();
+    }
+
+    private static void printlnWithIndent(String line) {
+        System.out.print("    ");
+        System.out.println(line);
+    }
+
+    /**
+     * Prints out the list of tasks added so far, or inform the user if no tasks have been added
+     * yet
+     */
+    private static void printTexts() {
+        if (!Rica.hasAnyTasks()) {
+            printlnWithIndent(" Hope I'm not amnesiac, but I don't remember any tasks?");
+        } else {
+            ArrayList<Task> tasks = Rica.getPastTexts();
+            printlnWithIndent(" I think you have these tasks:");
```

Comment:
Hardcoded strings can be refactored into constants at the top of code for maintainability